src/occupational_classification/lookup/soc_lookup.py

```diff
@@ -0,0 +1,277 @@
+    ) -> list[dict[str, Any]]:
+        """Retrieve unique code divisions from SOC candidates.
+
+        Returns:
+            list: Major group metadata.
+        """
+        unique_major_group: dict[str, dict[str, Any]] = {}
+
+        for candidate in soc_candidates:
+            major_group_info = self.lookup_code_major_group(candidate["soc_code"])
+            code_major_group = major_group_info["code_major_group"]
+
+            # Only add unique divisions
+            if (
+                isinstance(code_major_group, str)
+                and code_major_group not in unique_major_group
+            ):
+                unique_major_group[code_major_group] = major_group_info
+
+        return list(unique_major_group.values())
+
+
+class SOCRephraseLookup:
+    """A class for performing rephrased lookups of SOC codes based on descriptions.
+
+    This class extends the functionality of SOC lookups by allowing for
+    rephrased or alternative descriptions to be matched to SOC codes.
+
+    Attributes:
+        rephrase_dict (dict[str, str]): A dictionary mapping rephrased descriptions
+            to their corresponding SOC codes.
+        meta (SocMeta): Metadata for SOC classifications.
+
+    Methods:
+        rephrase_lookup(description: str) -> dict[str, Any]:
+            Looks up an SOC code based on a rephrased description.
+        add_rephrase_mapping(original: str, rephrased: str) -> None:
+            Adds a new rephrase mapping to the lookup dictionary.
+    """
+
+    def __init__(
+        self,
+        data_path_structure: str = "../src/occupational_classification/data/soc2020volume1structureanddescriptionofunitgroupsexcel16042025.xlsx",
+        data_path_index: str = "../src/occupational_classification/data/soc2020volume2thecodingindexexcel16042025.xlsx",
+    ):
+        self.data_path_structure = data_path_structure
+        self.data_path_index = data_path_index
+        self.data: pd.DataFrame = self.access_hierarchy()
+
+        # Create a lookup dictionary for quick access
+        self.lookup_dict: dict[str, str] = self.data.set_index("soc_code")[
+            "input_description"
+        ].to_dict()
+
+    def access_hierarchy(self):
+        """Converts the data for useful format for lookup method.
+
+        Returns:
+            pd.DataFrame: A DataFrame containing data useful
+            for rephrased lookups.
+        """
+        soc_structure = load_soc_structure(self.data_path_structure)
+        soc_df = SocDB.create_soc_dataframe(soc_structure)
+
+        soc_index = load_soc_index(self.data_path_index)
+
+        data = load_hierarchy(soc_df, soc_index).all_leaf_text()
+
+        data = data.rename(columns={"code": "soc_code", "text": "input_description"})
+
+        return data
+
+    def lookup(self, soc_code: Union[str, int]) -> dict[str, Union[str, Any]]:
+        """Retrieve reviewed description for the given SOC code."""
+        soc_code = str(soc_code)
```

Comment:
You allow both int and str, but then convert to str, is it because input format can not be checked?